src/tests/integration/test-data/cse-lemmas.k

```diff
@@ -30,9 +30,20 @@ module CSE-LEMMAS
       requires notBool ( lengthBytes(B1) ==Int lengthBytes(B2) )
       [simplification]
 
+    // Function selector does not match
+    rule { B:Bytes #Equals B1:Bytes +Bytes B2:Bytes } => #Bottom
+      requires #range(B, 0, 4) =/=K #range (B1, 0, 4)
+      [simplification(60), concrete(B, B1)]
+
+    // Function selector matches
+    rule { B:Bytes #Equals B1:Bytes +Bytes B2:Bytes } => {#range(B, 4, lengthBytes(B) -Int 4) #Equals B2}
+      requires #range(B, 0, 4) ==K B1
```

Comment:
```suggestion
      requires 4 <=Int lengthBytes(B) andBool #range(B, 0, 4) ==K B1
```

I have forgotten this constraint, otherwise `#range` will pad with zeros and equality is still possible.